### PR TITLE
Implementing `switch` expressions in the `cupertino/` directory

### PR DESCRIPTION
**Pull request by nate-thegrate**
This pull request is based on issue #136139 (it's a continuation of #139048 and #139882, this time focusing on the `packages/flutter/lib/src/cupertino/` directory.)

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by justinmc — APPROVED** [review]
LGTM 👍 

Thank you for doing this cleanup! I'm convinced this is more readable, and you saved 58 lines in the process.

Have you requested a test exemption?

File: packages/flutter/lib/src/cupertino/context_menu.dart [hunk 1]
```diff
@@ -904,14 +904,11 @@ class _ContextMenuRoute<T> extends PopupRoute<T> {
...
-        return AlignmentDirectional.topCenter;
-      case _ContextMenuLocation.right:
-        return AlignmentDirectional.topEnd;
-      case _ContextMenuLocation.left:
-        return AlignmentDirectional.topStart;
-    }
+    return switch (contextMenuLocation) {
+      _ContextMenuLocation.center => AlignmentDirectional.topCenter,
+      _ContextMenuLocation.right  => AlignmentDirectional.topEnd,
+      _ContextMenuLocation.left   => AlignmentDirectional.topStart,
```
[the remark this answers is not in the document]
Comment (justinmc) [reply packages/flutter/lib/src/cupertino/context_menu.dart:910]: And I just ran into one where we do align them:

https://github.com/flutter/flutter/blob/5aa6cb857d17df5073a4f5800a0a196a91ad2cc5/packages/flutter/lib/src/widgets/editable_text.dart#L4131-L4135

Based on the styleguide, I'd say we should standardize on aligning the arrows. So no change needed by you here. 👍

**Comment by nate-thegrate** [comment]
> LGTM 👍
> 
> Thank you for doing this cleanup! I'm convinced this is more readable, and you saved 58 lines in the process.
> 
> Have you requested a test exemption?

@justinmc Thank you very much!

I originally had a massive PR that implemented `switch` expressions throughout the entire repo, but it was recommended that I break it up into multiple smaller ones that are easier to review.

This is the third PR I've made; #139048 and #139882 didn't need a test exemption, so IMO we're good to merge this one as well.

**Comment by nate-thegrate** [comment]
Sweet, all tests have passed!

@justinmc could you add the "autosubmit" label when you have a chance?

File: packages/flutter/lib/src/cupertino/context_menu.dart [hunk 2]
```diff
@@ -1223,20 +1220,14 @@ class _ContextMenuRouteStaticState extends State<_ContextMenuRouteStatic> with T
...
-      case _ContextMenuLocation.right:
-        return orientation == Orientation.portrait
-          ? Alignment.bottomCenter
-          : Alignment.topLeft;
-      case _ContextMenuLocation.left:
-        return orientation == Orientation.portrait
-          ? Alignment.bottomCenter
-          : Alignment.topRight;
-    }
+    return switch ((contextMenuLocation, orientation)) {
```
Comment (LongCatIsLooong) [inline packages/flutter/lib/src/cupertino/context_menu.dart:1223]: nit: would it be more obvious that portrait is special if it's written this way?
```dart
(center || left || right, portrait) => Alignment.bottomCenter, 
 (_ContextMenuLocation.center, Orientation.landscape) => ...,
```

File: packages/flutter/lib/src/cupertino/date_picker.dart [hunk 3]
```diff
@@ -1757,13 +1757,10 @@ class _CupertinoDatePickerMonthYearState extends State<CupertinoDatePicker> {
...
 
     for (int i = 0; i < columnWidths.length; i++) {
-      late final double offAxisFraction;
-      switch (i) {
-        case 0:
-          offAxisFraction = -0.3 * textDirectionFactor;
-        default:
-          offAxisFraction = 0.5 * textDirectionFactor;
-      }
+      final double offAxisFraction = switch (i) {
```
Comment (LongCatIsLooong) [inline packages/flutter/lib/src/cupertino/date_picker.dart:1760]: The switch expression won't be able to provide exhaustiveness feedback for int values anyways. Consider using the ternary operator.

**Review by LongCatIsLooong — APPROVED** [review]

File: packages/flutter/lib/src/cupertino/context_menu.dart (hunk 2 above)
Comment (nate-thegrate) [reply packages/flutter/lib/src/cupertino/context_menu.dart]: Just refactored it to use an if-statement. Hopefully it's more readable now!

File: packages/flutter/lib/src/cupertino/date_picker.dart (hunk 3 above)
Comment (nate-thegrate) [reply packages/flutter/lib/src/cupertino/date_picker.dart]: done, thanks!

(I just did some other refactoring here too; hopefully it all looks good.)

**Pull request merged**